components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/ApplicationDAO.java

```diff
@@ -138,7 +138,7 @@ default ApplicationBasicInfo getApplicationBasicInfoByResourceId(String resource
         return null;
     }
 
-    default ServiceProvider addApplication(ServiceProvider application, String tenantDomain)
+    default String addApplication(ServiceProvider application, String tenantDomain)
```

Comment:
This might break the compatibility of the consumers

Reply:
This API was introduced specifically for application CRUD operations based on resourceIds. The only current consumer are the methods consumed by the REST API. Therefore we can change this API. Might need to bump the framework version by a minor version through